### PR TITLE
Changed Typos and Code in .cpp

### PR DESCRIPTION
**Pull request by MrKrishnaAgarwal**
@geoffromer 
please review this pr

### CODE REVIEW

**Comment by MrKrishnaAgarwal** [comment]
test / test (macos-latest, opt) (pull_request_target)

**why is this check failing?**

File: common/check_internal.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ class ExitingStream {
   [[noreturn]] ~ExitingStream() {
     llvm_unreachable(
         "Exiting streams should only be constructed by check.h macros that "
-        "ensure the special operator| exits the program prior to their "
+        "ensure the `|` operator exits the program prior to their "
```
Comment (jonmeow) [inline common/check_internal.h:40]: I don't think this should be changed; `operator|` is the function name in C++.

File: explorer/syntax/lex_scan_helper.cpp [hunk 2]
```diff
@@ -22,9 +22,9 @@ auto StringLexHelper::Advance() -> bool {
   return true;
 }
 
-auto ReadHashTags(Carbon::StringLexHelper& scan_helper,
-                  const size_t hashtag_num) -> bool {
-  for (size_t i = 0; i < hashtag_num; ++i) {
+auto ReadHashTags(Carbon::StringLexHelper& scan_helper, const int hashtag_num)
```
Comment (jonmeow) [inline explorer/syntax/lex_scan_helper.cpp:25]: We'll sometimes use size_t when it seems appropriate: https://google.github.io/styleguide/cppguide.html#Integer_Types

Perhaps most importantly though, it's size_t in the header and these shouldn't diverge.

File: docs/spec/lang/README.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
     same library and the same [scope](names.md#scopes) and declare the same
     [name](names.md).
 
-    TODO: Linkage rules for foreign entities. TODO: Ability to declare
+    TODO: Linkage rules for foreign entities. <br> TODO: Ability to declare
```
Comment (jonmeow) [inline docs/spec/lang/README.md:69]: I think it's fine to have multiple TODOs here; we shouldn't use HTML in md files unless absolutely necessary anyways.

**Review by jonmeow — CHANGES_REQUESTED** [review]
I'm going to close this PR. I've commented on each proposed change to explain why.

**Comment by jonmeow** [comment]
> test / test (macos-latest, opt) (pull_request_target)
> 
> **why is this check failing?**

For future reference, if you click "details" next to the check, you can read the output from the action.

**Comment by MrKrishnaAgarwal** [comment]
Okay Thanks 
No Worries 👍

**Comment by jonmeow** [comment]
@MrKrishnaAgarwal As I'm reviewing more PRs, I see your change to check_internal.h seems to be identical to that in #1699 and lex_scan_helper.cpp is identical to that in #1757. Please make sure you aren't accidentally including changes being made by other developers. (more an issue for future PRs than this)

**Comment by MrKrishnaAgarwal** [comment]
Oh!

I didn't noticed that 
But can I make another PR with the changes of my previous pull request that closed unexpectedly. #1693 

@jonmeow

**Pull request closed without merging**